src/libMediator/Mediator.cpp

```diff
@@ -242,3 +243,19 @@ void Mediator::IncreaseEpochNum() {
 }
 
 bool Mediator::GetIsVacuousEpoch() { return m_isVacuousEpoch; }
+
+uint32_t Mediator::GetShardSize(const bool& useShardStr) const {
+  uint32_t shardNodeNum = 0;
+
+  if (useShardStr) {
+    for (const auto& shard : m_ds->m_shards) {
+      shardNodeNum += shard.size();
+    }
+  } else {
+    shardNodeNum = m_ds->m_allPoWs.size();
+  }
+
+  return (shardNodeNum < 651)
```

Comment:
I think maybe we can move this decision-making logic into `CalculateShardSize`, so we don't use the magic number 651 in two places.